purchase_quick/models/product_product.py

```diff
@@ -19,21 +19,34 @@ class ProductProduct(models.Model):
         help="Technical: used to compute quantities to purchase.",
     )
 
+    def _default_quick_uom_id(self):
+        if self.env.context.get("parent_model", False) == "purchase.order":
+            return self.uom_po_id
+        return super()._default_quick_uom_id()
+
+    def _compute_quick_uom_category_id(self):
+        if self.env.context.get("parent_model", False) == "purchase.order":
+            for rec in self:
+                rec.quick_uom_category_id = rec.uom_po_id.category_id
+        else:
+            super()._compute_quick_uom_category_id()
+
+    def _compute_process_qty_purchase(self):
+        po_lines = self.env["purchase.order.line"].search(
+            [("order_id", "=", self.env.context.get("parent_id"))]
+        )
+        for product in self:
```

Comment:
@PierrickBrun @kevinkhao 
I think we have an issue here (the issue was here before).
We sum the qty of product for each line. But them we write the total qty in the first line.

So in the case I have 2 PO line with product FOO:
- 10 x FOO
- 20 x FOO

I will have 30 FOO in the product qty, If I set 35 I will have on my PO
- 35 FOO
- 20 FOO

In order to keep this module simplier, and also stick to real case.
I think we should keep the sum here BUT add a raise in _get_quick_line if we have several line.

In real word we should never have twice the same product in a PO and If it's the case, for example with purchase_lot, it doesn't make sense to use mass addition.

So let's keep it simple and not broken

Reply:
done

Reply:
done